drift-client/src/test/java/io/airlift/drift/client/address/SimpleAddressSelectorBinderTest.java

```diff
@@ -0,0 +1,122 @@
+    private static final String PREFIX = "testService";
+
+    @Test
+    public void testNoDefaults()
+            throws Exception
+    {
+        ImmutableList<HostAndPort> addresses = ImmutableList.of(HostAndPort.fromParts("example.com", 1), HostAndPort.fromParts("example.com", 2));
+        ImmutableMap<String, String> configurationProperties = ImmutableMap.of(PREFIX + ".thrift.client.addresses", "example.com:1,example.com:2");
+        testAddressSelector(simpleAddressSelector(), configurationProperties, addresses);
+    }
+
+    @Test
+    public void testSingleDefault()
+            throws Exception
+    {
+        HostAndPort address = HostAndPort.fromParts("example.com", 1);
+        testAddressSelector(simpleAddressSelector(address), ImmutableMap.of(), ImmutableList.of(address));
+
+        testAddressSelector(simpleAddressSelector(address),
+                ImmutableMap.of(PREFIX + ".thrift.client.addresses", "example.com:11,example.com:22"),
+                ImmutableList.of(HostAndPort.fromParts("example.com", 11), HostAndPort.fromParts("example.com", 22)));
+    }
+
+    @Test
+    public void testMultipleDefaults()
+            throws Exception
+    {
+        ImmutableList<HostAndPort> addresses = ImmutableList.of(HostAndPort.fromParts("example.com", 1), HostAndPort.fromParts("example.com", 2));
+        testAddressSelector(simpleAddressSelector(addresses), ImmutableMap.of(), addresses);
+
+        testAddressSelector(simpleAddressSelector(addresses),
+                ImmutableMap.of(PREFIX + ".thrift.client.addresses", "example.com:11,example.com:22"),
+                ImmutableList.of(HostAndPort.fromParts("example.com", 11), HostAndPort.fromParts("example.com", 22)));
+    }
+
+    private static void testAddressSelector(
+            AddressSelectorBinder addressSelectorBinder,
+            Map<String, String> configurationProperties,
+            List<HostAndPort> expected)
+    {
+        Bootstrap app = new Bootstrap(
+                new ThriftCodecModule(),
+                binder -> addressSelectorBinder.bind(binder, THRIFT_SERVICE_ANNOTATION, PREFIX));
+
+        LifeCycleManager lifeCycleManager = null;
+        try {
+            Injector injector = app
+                    .setRequiredConfigurationProperties(configurationProperties)
+                    .strictConfig()
+                    .doNotInitializeLogging()
+                    .initialize();
+            lifeCycleManager = injector.getInstance(LifeCycleManager.class);
+
+            AddressSelector<?> addressSelector = injector.getInstance(Key.get(AddressSelector.class, THRIFT_SERVICE_ANNOTATION));
+            assertTrue(addressSelector instanceof SimpleAddressSelector);
+            SimpleAddressSelector simpleAddressSelector = (SimpleAddressSelector) addressSelector;
+            assertEquals(simpleAddressSelector.getAddresses(), expected);
+        }
+        catch (Exception e) {
```

Comment:
All the test methods throw `Exception`, so you can remove this and have it propagate